vocab-builder-maven-plugin/src/main/java/com/github/tkurz/sesame/vocab/plugin/VocabularyBuilderMojo.java

```diff
@@ -245,19 +313,50 @@ public void execute() throws MojoExecutionException, MojoFailureException {
                         target = target.resolve(builder.getPackageName().replaceAll("\\.", "/"));
                         Files.createDirectories(target);
                     }
+                    if ( createUriConstants && uriGeneration != null ) {
```

Comment:
Nope, it does actually negate the global safety. You need to reorganise this code and document why the merge would not happen with vocab, but it would happen in this case. Naively, although it is hard to follow your code, I would have thought you would always want to merge the global settings into the vocab specific settings.